src/resources/app/components/Container/AnnouncementContainer.vue

```diff
@@ -20,17 +23,18 @@ import CreateAnnouncement from "../CreateAnnouncement";
 import AnnouncementCard from "../AnnouncementCard";
 import api from "../../../api/announcement/GetAnnouncements";
 
-const courseId = window.location.href.split("/")[4];
-
 export default {
   name: "AnnouncementContainer",
   data() {
     return {
       announcements: []
     }
   },
+  props:[
+  "courseId",
+  ],
   methods: {
-    async getAnnouncements() {
+    async getAnnouncements(courseId) {
       try {
         const response = await api.GetAnnouncements(courseId);
```

Comment:
My bad, I missed this one before. It is a bit weird to use PascalCase for a function name. Please consider changing it to camelCase if you have time

Reply:
Noted, I already changed mine. For the other function such as CreateAnnouncement, Edit, and Get will be refactor by @FXDROS